### DISCUSSION TITLE
Q/FR: Make certain alerts work after some time while another (alerts work) immidiately

### DISCUSSION THREAD

**Discussion by ARkaine2** [NightscoutFoundation/xDrip#2542] [General]
Is it possible to make certain alerts work after some time while another (alerts work) immidiately?..

Let's say:
Hyper alert 1 BG 8 work after 60 minutes (delay/snooze) (when BG is 8> 10<)
Hyper alert 2 BG 10 work immidiately

The problem of using "Start Snoozed" (or something like that) is that:
If to set up time I'm interested in to "delay" alert (let's say 0 -- but it's not possible, or at least 10 minutes for hyper alert 2 with BG 10),
I'll loose "regular" time for snooze function with volume button.

**Comment by Navid200** [comment]
If you are talking about the snooze time, each alert already has a specific snooze time and you can edit each separately.

**Reply by ARkaine2** [reply]
I'm talking about something like "Start Snoozed" but with keeping "regular"/different time snooze (with volume button) working.

And is it even possible to set 0 snooze time if make alerts working as "Start Snoozed"?

**Reply by Navid200** [reply]
Why don't you try it?

**Reply by ARkaine2** [reply]
I tried, 10 mins -- minimal snooze value. 
(so, if make alerts "start snoozed" it always will have at least 10 mins delay -- what is not preferable at higher BG alerts -- what I'm talking about)

I'm asking about 0 mins -- is there any way to make it? Seems that by default it's not possible.

And if set alerts snooze time "specified" for "start snoozed", we'll loose "default" snooze time/function 

.

Let's say I want:

Function "Start Snoozed" is on

Hyperglycemia 1 alert with BG 8 has snooze time 60 minutes and it works after 60 minutes with BG 8>.
^
All is fine with it.

Hyperglycemia 2 alert with BG 10 work immediately
^
Problem 1: it always have at least 10 minutes snooze ("delay").
Problem 2: I want to keep possibility to snooze it (not in "snooze menu", but with volume button -- one click) for certain time (let's say 60 minutes), but if 10 mins are specified as default snooze time (to make required "delay"), I'll have only 10 mins snooze (with volume button).

**Reply by Navid200** [reply]
Thanks for testing that.
You are helping me understand what you need.

Can you now tell me what will happen if you disable start snoozed? 

Please note that when you make a long post, it may be hard for someone who doesn't know anything to read all of it.

**Reply by ARkaine2** [reply]
> Can you now tell me what will happen if you disable start snoozed?

All alerts start immediately.

**Reply by Navid200** [reply]
And what's wrong with that?

Do you use persistent high alert?

**Reply by Navid200** [reply]
If I am really really high, I would like to know immediately.  That's why I have disabled start snoozed.

If I am just a little bit high, it is very like that may be because I have just had food and insulin is trying to catch up.  So, I don't want to get an alert right away.  But, if it persists, I get an alert because I use persistent high alert.

There has been a request to allow multiple persistent high alerts.  I have to say thinking about that request, I see nothing wrong with it.   But, this request is something that needs to be thought of very carefully by multiple developers and discussed before added.  So, I have left it alone because there has not been multiple requests.

**Reply by ARkaine2** [reply]
> And what's wrong with that?
> 
> Do you use persistent high alert?

That's nice idea... I need to think about it... But...

So I can set up:

"Start Snoozed" is on.

Hyperglicemia alert 1 (BG8) with 1 hour "delay" (snooze).
and
Hyperglicemia alert 2 (BG10) with 10 minutes "delay" (snooze).

Seems that problem will be AGAIN that it's not possible to snooze Hyperglicemia alert 2 with one click (volume button) for longer time than 10 minutes.

Or does persistent high alert "make silent" (let's say for 1 hour) alert (2) with snooze time 10 minutes?

**Reply by Navid200** [reply]
I don't follow.

Any alert can be snoozed by pressing the volume button when the phone has been unlocked if you have enabled "Buttons silence alarms" setting.
Are you saying it doesn't work?

**Reply by ARkaine2** [reply]
Excactly, you understand the request/problem. :)

> If I am just a little bit high, it is very like that may be because I have just had food and insulin is trying to catch up. So, I don't want to get an alert right away. But, if it persists, I get an alert because I use persistent high alert.

So, how / is it possible to make "just a little bit high" alert with "disabled start snoozed" delayed/snoozed/persistent or that was your developer thoughts?

**Reply by Navid200** [reply]
The persistent high alert only triggers if you have been higher than your High level for a certain amount of time.  
The level of the persistent high alert is the same as your High level.
How long you must be higher before the alert triggers is one of the settings of the persistent high alert.

**Reply by ARkaine2** [reply]
> Any alert can be snoozed by pressing the volume button when the phone has been unlocked if you have enabled "Buttons silence alarms" setting.
> Are you saying it doesn't work?

They work fine,
Maybe my "logic" is wrong, but 
I see problem in:

If snooze by pressing the volume button
and use "start snoozed" function

We can want different time for:
snooze by pressing the volume button 
and snooze for "start snoozed" -- which is more like delay/wait (of alert first sounding).

**Reply by Navid200** [reply]
It's not easy to do this.
I thought you said you could not snooze using the volume button.  Now you say you can.

I am not sure what is not working.
I am doing my best here.  

xDrip already has too many options.  If I had my way, I would remove half of them altogether!
Adding more options is a last resort.  We will not add more options unless absolutely necessary.

Tell me what is not working.  Please be clear.  and please don't change the subject.  If you can show me what doesn't work, I will open an issue and see to it the the right developer sees it.

**Reply by ARkaine2** [reply]
> I am doing my best here.

I understand that. Sorry for troublemaking.

.

> There has been a request to allow multiple persistent high alerts. I have to say thinking about that request, I see nothing wrong with it. But, this request is something that needs to be thought of very carefully by multiple developers and discussed before added. So, I have left it alone because there has not been multiple requests.

Your message about multiple persistent high alerts is closest to what I'm asking about.
But I think that I need multiple start snoozed time (from 0) for different alerts with keeping "regular" (maybe 1 overall, or maybe multiple as ATM) snooze time for each alert that will be snoozed by volume button.
Probably you, as developer, can see it clearier.

.

My "quick idea" (which can be extremely difficult to realize and which probably will extremely complicate all) is to:
Make "different" snooze time settings for:

Snooze ("delay"/"wait") in each alert's settings, which will work with "start snoozed" function.

[screenshot 1](https://user-images.githubusercontent.com/31447095/206565532-8df9e5ca-ac9d-4865-af80-ea8974241f25.png)

And for 

Snooze "as snooze" -- with volume button or from "special" (snooze) menu (which is in my opinion a little bit empty). 

[screenshot 2](https://user-images.githubusercontent.com/31447095/206565559-eae630be-0476-49d4-aff2-928f4f6add67.png)

.

> If I had my way, I would remove half of them altogether!

Hehe. Same here. Most of what I need is "advanced" alerts system. :)
But probably different people can want different functionality.

**Reply by Navid200** [reply]
You may be able to accomplish your objectives using xDrip as is.  
Give persistent high alert a try.

After you do that, we can talk about what you think is missing.

Let me give you an analogy:
You contact a car manufacturer and tell them that you want to be able  to move the gas and break pedals back and forth because different people drive the car and they have different body dimensions.
The car manufacturer tells you that you can already move the seat back and forth.  That effectively allows you to accomplish the same objective.

You say "but, that's not what I want".

What do you expect the car manufacturer to tell you?

**Reply by ARkaine2** [reply]
I answered about persistent high alert, 
but I'll (try to) make my answer more detail/clear:

If (with "Persistent High Alert") "Start Snoozed" is off -- then ANY ALERT (let's say on BG 8) starts immediately -- that's the problem.

If (with "Persistent High Alert") "Start Snoozed" is on -- then as i described the problem earlier:

For example:

Hyperglicemia alert 1 (BG8) with 60 minutes snooze ("delay"/"wait").

and

Hyperglicemia alert 2 (BG10) with 10 minutes snooze ("delay"/"wait").

^

Problem is that it's not possible to snooze alert 2 with volume button for longer time than 10 minutes.

.

I need as I secribed earlier:
If BG is 8> for 60 minutes, it makes alert.
If BG is 10> it makes alert immediately.
I can snooze (with volume button) BG 10> alert for example for 60 minutes.

**Reply by ARkaine2** [reply]
In other words, I think, that:

Starting "delay"/"wait", 
— which ATM/by default is realized by "snooze" in alert settings (which works with "start snoozed"). 

AND 

"Real" snooze 
— that we make with volume button or with certain menu. 

^

"SHOULD" BE DIFFERENT SETTINGS.

And that's no such complication (at least for user) because:

1 thing is "delay"/"wait" for alert start. 
2 thing is alert snooze.

Maybe multiple persistent high alerts will partically cover this reqest (so, your suggestion is close), but I see ahead troubles with snoozing with it.

.

Let's say even if we have multiple persistent high alerts. 

Then 

If start snoozed is off — any alert (even that one which should be delayed) will work immidiately — which not covers my (and others) request.

If start snoozed is on —

Let's say we have 2 hyper alerts:

Hyper 1, BG 8, persistent high alert any value, let's say 60 minutes, 

At least we have to set snooze time higher than persistent high alert time, because in other case we'll have alert that will work earlier and "all" (request/idea of snoozed alert start) will collapse.

Hyper 2 (don't forget that in example start snoozed is on, because we have to, look earler), BG 10, persistent high alert any value, snooze is minimal to make it work immidiately (by default it's 10 minutes, I'd prefer to make 0 possible).

And the problem in this case that snooze in this alert's settings is minimal to make it work immidiately — so, if we'll use volume button to snooze this alert, it will snooze it for short period of time (10 mins). 

.

I see simplest fix of this problem and satisfaction of request (and some nearest needs) as described earlier:

1 Make "delay"/"wait" in each alert's settings. 
2 Make multiple snooze settings (for each alert that will be) displayed in snooze menu. 

But probably you as (co)developer see it clearier.

Thanks.

**Reply by Navid200** [reply]
My suggestion is to:
1- Disable start snoozed.
2- Keep your high alerts that are for very high levels.
3- Temporarily, disable high alerts that are for not very high values.
4- Enable persistent high alert.

Play with that and tweak it to make it do what you need.
Only after you have gained experience with persistent high alert, if you think xDrip is missing something, I will be all ears.

**Reply by ARkaine2** [reply]
Thanks for your sincere attempt to help. :)

I got your idea about persistent high alerts... And maybe your advice is better, 
But:
I wouldn't like to loose "intermediate" hyper alerts.
And statistics probably will became worse, if I'll set hyper let's say at 8 or 9 mmol.

So

I almost found succes with start snoozed on.
(start snoozed is on, because if it's off -- alerts will sound without delay/wait).

So, ATM I have:
Start Snoozed On
"Pre"-Hyper Alert, BG 9, snooze time 90
Hyper Alert 1, BG 11, snooze time 10 mins
Hyper Alert 2, BG 13, snooze time 10 mins

Only one trouble of this is:

If I'll try to snooze hyper alert 1 or 2 (BG 11 or 13) with volume button -- it will snooze it only for 10 minutes (because it has snooze time 10 mins).
I'd prefer to snooze it with volume button for longer time, like if it has 60--90 minutes in snooze settings.
But I need to work this alert imidiately. ¯\_(ツ)_/¯

And one more thing:

Persistent high alert is "untouchable" (it's not possible to affect how long it will continue except time in settings).
With snooze by hand (volume button) I can continue snooze at any moment -- so with persistent high alert this "function" will be lost.

Anyway real thanks for your advice, it helped me.
